### PR TITLE
ACPI: add rt1308 acpi on sdw bus 2

### PR DESCRIPTION
**Pull request by bardliao**
Add rt1308 information on SdW bus 2.

Signed-off-by: Bard Liao <yung-chuan.liao@linux.intel.com>

### CODE REVIEW

File: SDW/alc1308.asl [hunk 1]
```diff
@@ -0,0 +1,587 @@
+ /** @file
+  The definition block in ACPI table for rt5682 device under I2C0 Controller
+
+Copyright (c) 2015, Intel Corporation. All rights reserved.<BR>
```
Comment (plbossart) [inline SDW/alc1308.asl]: 2019

File: SDW/alc1308.asl [hunk 2]
```diff
@@ -0,0 +1,587 @@
+ /** @file
+  The definition block in ACPI table for rt5682 device under I2C0 Controller
```
Comment (plbossart) [inline SDW/alc1308.asl]: rt1308 under SoundWire controller

File: SDW/alc1308.asl [hunk 3]
```diff
@@ -0,0 +1,587 @@
...
+                        Zero
+                    }, 
+
+                    Package (0x02)
+                    {
+                        "mipi-sdw-port15-read-behavior", 
+                        One
+                    }, 
+
+                    Package (0x02)
```
Comment (plbossart) [inline SDW/alc1308.asl]: remove this, it makes no sense

File: SDW/alc1308.asl [hunk 4]
```diff
@@ -0,0 +1,587 @@
...
+                ToUUID ("dbb8e3e6-5886-4ba6-8795-1319f52a966b"), 
+                Package (0x0E)
+                {
+                    Package (0x02)
+                    {
+                        "intel-endpoint-descriptor-0", 
+                        "EPD0"
+                    }, 
+
+                    Package (0x02)
```
Comment (plbossart) [inline SDW/alc1308.asl]: also remove all the link0..3 properties, they are wrong.

File: SDW/alc1308.asl [hunk 5]
```diff
@@ -0,0 +1,587 @@
...
+                    {
+                        "mipi-sdw-dp-6-source-subproperties", 
+                        "DPN"
+                    }, 
+
+                    Package (0x02)
+                    {
+                        "mipi-sdw-dp-7-sink-subproperties", 
+                        "DPN"
+                    },
```
Comment (plbossart) [inline SDW/alc1308.asl]: that's a lot of ports, are they all valid?

File: SDW/alc1308.asl [hunk 6]
```diff
@@ -0,0 +1,587 @@
...
+  0x1000
+  )
+{
+External(\_SB.PCI0.HDAS.SNDW, DeviceObj)
+External(SDW2, UnknownObj)
+
+Scope (_SB.PCI0.HDAS.SNDW) {
+	Device (RTKX)
+        {
+            Name (_ADR, 0x000220025D130800)  // _ADR: Address
```
Comment (plbossart) [inline SDW/alc1308.asl]: you sure about the _ADR?
should be link 2, rtk ID is 025D and part 1308 so there's an extra 2?
also the links start at zero, link2 is the 3rd master.

**Review by plbossart — CHANGES_REQUESTED** [review]
thanks @bardliao couple of points below to look at.

We may want to add also the ALC711 on link0 while we are at it, and we need two instances for RT308.

File: SDW/alc1308.asl (hunk 5 above)
Comment (bardliao) [reply SDW/alc1308.asl]: @plbossart I am not sure about it, will check with Realtek.

File: SDW/alc1308.asl (hunk 6 above)
Comment (bardliao) [reply SDW/alc1308.asl]: > you sure about the _ADR?
> should be link 2, rtk ID is 025D and part 1308 so there's an extra 2?

@plbossart The first 2 is sdw version, should we remove it?

> also the links start at zero, link2 is the 3rd master.

Yes, it is connected to 3rd master on my platform.

**Comment by bardliao** [comment]
> We may want to add also the ALC711 on link0 while we are at it, and we need two instances for RT308.

@plbossart I will update it once I can test ALC711 and two ALC1308 :)

**Comment by plbossart** [comment]
On 8/6/19 5:12 AM, Bard Liao wrote:
>     We may want to add also the ALC711 on link0 while we are at it, and
>     we need two instances for RT308.
> 
> @plbossart <https://github.com/plbossart> I will update it once I can 
> test ALC711 and two ALC1308 :)

there's no harm in adding them upfront, you can always filter the links 
and if there is no driver then there is no match and no probe.

File: SDW/rtk_sdw.asl [hunk 7]
```diff
@@ -0,0 +1,1970 @@
...
+  "RTKSDWTabl",
+  0x1000
+  )
+{
+External(\_SB.PCI0.HDAS.SNDW, DeviceObj)
+
+Scope (_SB.PCI0.HDAS.SNDW) {
+	Device (RTK0)
+        {
+            Name (_ADR, 0x000020025D071100)  // _ADR: Address
```
Comment (RanderWang) [inline SDW/rtk_sdw.asl]: I tested it on ICL with 3-in-1 board, it should be 0x000021025D071100

File: SDW/rtk_sdw.asl [hunk 8]
```diff
@@ -0,0 +1,1970 @@
...
+                    {
+                        "mipi-sdw-audio-mode-glitchless-transitions",
+                        Zero
+                    }
+                }
+            })
+        }
+	Device (RTK2)
+        {
+            Name (_ADR, 0x000220025D130800)  // _ADR: Address
```
Comment (RanderWang) [inline SDW/rtk_sdw.asl]: it should be 0x000222025D130800

**Review by RanderWang — COMMENTED** [review]
Please refine it

**Comment by plbossart** [comment]
replaced by #6

**Pull request closed without merging**